src/main/java/sysc4806/graduateAdmissions/model/Privilege.java

```diff
@@ -0,0 +1,39 @@
+package sysc4806.graduateAdmissions.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+/**
+ * This class is used to specify privileges for roles. A Role with have a collection
+ * of these objects to determine what it can and cannot do. A Privilege is made up of three
+ * parts: An action (eg. create, delete), a target for that action (eg. application, interest),
+ * and an owner of that target (eg. self, everyone). Combinations of these allow the
+ * specification of many different privileges for roles.
+ *
+ * For example, a student would have privileges for creating and deleting their own applications.
+ * Admin staff would have privileges for creating and deleting any student's application.
+ *
+ * @author luke
+ */
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Privilege {
+    //the primary key for a privilege
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @EqualsAndHashCode.Exclude
+    private long id;
+    //the type of CRUD operation the privilege specified
+    private Operation operation;
+    //the type of object that an operation is performed on
+    private Target target;
+    //the owner of the specified target object of the specified operation
+    private Owner owner;
+}
```

Comment:
I'm wondering how we're going get all of a user's privileges. I'm not sure if this is out of scope of this issue since theres not much written in the issue itself, but it looks like we'll need to create another class (enum?) that builds a list of everyone's Privilege objects - something we can grab when we start serving our api.

When we log in, we would want to hit an endpoint to get the user's info which will also getPermissions(user's roleName).

Reply:
a User object has one or more Role objects, a Role has multiple Privilege objects. This is not in the scope of this issue, but will become apparent to you if you look at each open issue for creating the model objects.